.github/workflows/build_wheels.yml

```diff
@@ -0,0 +1,111 @@
+name: Build and test
+
+on:
+  # Trigger on pull requests.
+  pull_request:
+
+  # Trigger on pushes to the mainline branches and version tags. This prevents building commits
+  # twice when the pull request source branch is in the same repository.
+  push:
+    branches:
+      - "maint"
+      - "master"
+    tags:
+      - "v*"
+
+  # Trigger on request.
+  workflow_dispatch:
+
+env:
+  SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, macos-10.15] #, windows-2019]
+
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          submodules: "recursive"
+
+      - name: Build wheels
+        uses: joerick/cibuildwheel@v1.10.0
+        env:
+          # Build for cpython >= 3.6.
+          CIBW_PROJECT_REQUIRES_PYTHON: ">=3.6"
+          CIBW_SKIP: pp*
+
+          CIBW_ENVIRONMENT_LINUX: "CMAKE_PREFIX_PATH=/project/tbb LD_LIBRARY_PATH=/project/tbb/lib/intel64/gcc4.8:$LD_LIBRARY_PATH"
+          CIBW_ENVIRONMENT_MACOS: "CMAKE_PREFIX_PATH=/Users/runner/work/freud/freud/tbb LD_LIBRARY_PATH=/Users/runner/work/freud/freud/tbb/lib/intel64/gcc4.8:$LD_LIBRARY_PATH"
+
+          # Set up submodules and TBB
+          CIBW_BEFORE_BUILD_LINUX: "source .github/workflows/cibuildwheel-before-build.sh {package}"
+          CIBW_BEFORE_BUILD_MACOS: "source .github/workflows/cibuildwheel-before-build.sh {package}"
+          CIBW_BEFORE_BUILD_WINDOWS: "powershell .github/workflows/cibuildwheel-before-build.ps1 {package}"
+
+          # Build only on 64-bit architectures.
+          CIBW_ARCHS_MACOS: x86_64
+          CIBW_ARCHS_LINUX: auto64
+          CIBW_ARCHS_WINDOWS: auto64
+
+          # Test the wheels.
+          CIBW_BEFORE_TEST_LINUX: "source .github/workflows/cibuildwheel-before-test.sh {package}"
+          CIBW_BEFORE_TEST_MACOS: "source .github/workflows/cibuildwheel-before-test.sh {package}"
+          CIBW_BEFORE_TEST_WINDOWS: "powershell .github/workflows/cibuildwheel-before-test.ps1 {package}"
+          CIBW_TEST_COMMAND: "cd {package}/tests && pytest . -v --log-level=DEBUG -k Box"
+
+#     - uses: actions/upload-artifact@v2.2.2
+#       with:
+#         path: ./wheelhouse/*.whl
+
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.4
+
+      - uses: actions/setup-python@v2.2.1
+        name: Install Python
+        with:
+          python-version: '3.9'
+
+      - name: Install build requirements
+        run: python3 -m pip --disable-pip-version-check install scikit-build cmake -r requirements.txt --upgrade --progress-bar=off
+
+      - name: Build sdist
+        run: python setup.py sdist
+
+#     - uses: actions/upload-artifact@v2.2.2
```

Comment:
I found a few issues when testing this part of the GSD `build_wheels.yml`. The latest version also adds TestPyPI uploads on non-tagged builds.

https://github.com/glotzerlab/gsd/blob/fdd624f65ce71a88057a4c8a5707c93c7fa604c8/.github/workflows/build_wheels.yml#L72-L115

Reply:
Thanks for the tip. I updated this section.